tests/builder.rs

```diff
@@ -112,3 +124,77 @@ fn bundle_chain() {
     // Verify the shielded bundle.
     verify_bundle(&shielded_bundle, &vk);
 }
+
+#[test]
+fn issue_bundle_chain() {
+    let mut rng = OsRng;
+    let pk = ProvingKey::build();
+    let vk = VerifyingKey::build();
+
+    let sk = SpendingKey::from_bytes([0; 32]).unwrap();
+    let fvk = FullViewingKey::from(&sk);
+    let recipient = fvk.address_at(0u32, Scope::External);
+
+    let isk = IssuerAuthorizingKey::from(&sk);
+    let ik = IssuerValidatingKey::from(&isk);
+
+    let asset_descr = "zsa_asset";
+
+    // Create a shielding bundle.
+    let mut unauthorized = IssueBundle::new(ik);
+
+    let note_type = unauthorized
+        .add_recipient(
+            asset_descr.to_string(),
+            recipient,
+            NoteValue::from_raw(5000),
+            false,
+            &mut rng,
+        )
+        .unwrap();
+
+    let sighash = unauthorized.commitment().into();
+    let proven = unauthorized.prepare(sighash);
+    let issuing_bundle = proven.sign(&mut rng, &isk).unwrap();
+
+    // Verify the shielding bundle.
+    verify_issue_bundle(&issuing_bundle);
+
+    // Create a shielded bundle spending the previous output.
+    let shielded_bundle: Bundle<_, i64> = {
+
+        // Take first note from first action
+        let note = issuing_bundle.actions().first().unwrap().notes().first();
+
+        // Use the tree with a single leaf.
+        let cmx: ExtractedNoteCommitment = note.commitment().into();
+        let leaf = MerkleHashOrchard::from_cmx(&cmx);
+        let mut tree = BridgeTree::<MerkleHashOrchard, 32>::new(0);
+        tree.append(&leaf);
+        let position = tree.witness().unwrap();
+        let root = tree.root(0).unwrap();
+        let auth_path = tree.authentication_path(position, &root).unwrap();
+        let merkle_path = MerklePath::from_parts(
+            u64::from(position).try_into().unwrap(),
+            auth_path[..].try_into().unwrap(),
+        );
+        let anchor = root.into();
+        assert_eq!(anchor, merkle_path.root(cmx));
+
+        let mut builder = Builder::new(Flags::from_parts(true, true), anchor);
+        assert_eq!(builder.add_spend(fvk, *note, merkle_path), Ok(()));
+        assert_eq!(
+            builder.add_recipient(None, recipient, NoteValue::from_raw(5000), note_type, None),
+            Ok(())
+        );
+        let unauthorized = builder.build(&mut rng).unwrap();
+        let sighash = unauthorized.commitment().into();
+        let proven = unauthorized.create_proof(&pk, &mut rng).unwrap();
+        proven
+            .apply_signatures(&mut rng, sighash, &[SpendAuthorizingKey::from(&sk)])
+            .unwrap()
+    };
```

Comment:
Let's test all 5 cases:
- trivial (already implemented)
- only non-native: 1 spend - 2 recipient notes
- only non-native: 2 spend - 1 recipient notes and padding 
- only non-native: 2 spend - 2 recipient notes with different denomination.
- mix with native notes.

Reply:
done